whittle/models/gpt/blocks/causal_self_attention.py

```diff
@@ -11,7 +11,7 @@
 
 
 class CausalSelfAttention(nn.Module):
-    def __init__(self, config: Config) -> None:
+    def __init__(self, config: Config, block_idx: int) -> None:
```

Comment:
should we pass some default parameter to `block_idx`?

Reply:
Yes the api of block and causal attention changes and this will be a breaking change